Dans Launcher terminal bash
se placer dans directory Projet-Methylation
cd /shared/projects/202304_duo/sletuve/Projet/Projet-Methylation
vérifier avec pwd

# Dépôt GitHub:
**création d'un dépôt**

Sur Github, en haut à droite, cliquer sur +

Puis nommer le dépôt (ici Projet-Methylation)

git@github.com:sevlet94/Projet-Methylation.git

et clone à partir de l'onglet 'Git' de Jypyter dans le dossier

se déplacer dans le dossier cloné: cd Projet-Methylation

**configuration du dépot local**

$ git config --global user.name "Séverine Létuvé"

$ git config --global user.email "severine.letuve@inserm.fr"

vérification avec $ git config --list | grep user

**configuration de Nano comme éditeur par défaut**

$ git config --global core.editor "nano"

Contenu du dossier
$ git status

**Ajout de NoteBookAnalyseMethylation.ipynb**
$ git add NoteBookAnalyseMethylation.ipynb
$ git commit -m "Premier commit"
$ git push

**Création README**
Sur site Github
puis récupérer sur local

$ pull

Lire contenu README.md
$ cat README.md

Compte-rendu de l'activité sur dépôt
$ git log

In [1]:
citation("minfi")


The minfi package is described in:

Aryee MJ, Jaffe AE, Corrada-Bravo H, Ladd-Acosta C, Feinberg AP, Hansen
KD, Irizarry RA (2014). “Minfi: A flexible and comprehensive
Bioconductor package for the analysis of Infinium DNA Methylation
microarrays.” _Bioinformatics_, *30*(10), 1363-1369. doi:
10.1093/bioinformatics/btu049 (URL:
https://doi.org/10.1093/bioinformatics/btu049).

SWAN normalization (preprocessSWAN) is described in:

Maksimovic J, Gordon L, Oshlack A (2012). “SWAN: Subset quantile
Within-Array Normalization for Illumina Infinium HumanMethylation450
BeadChips.” _Genome Biology_, *13*(6), R44. doi:
10.1186/gb-2012-13-6-r44 (URL:
https://doi.org/10.1186/gb-2012-13-6-r44).

Funnorm normalization (preprocesFunnorm) is described in:

Fortin J, Labbe A, Lemire M, Zanke BW, Hudson TJ, Fertig EJ, Greenwood
CM, Hansen KD (2014). “Functional normalization of 450k methylation
array data improves replication in large cancer studies.” _Genome
Biology_, *15*(12), 503. doi: 10.1186/s13059-

# Utilisation du package Minfi pour l'analyse de puces à méthylation 450K Illumina (Illumina’s Methylation arrays 450K and Epic)



**User's guide**
https://bioconductor.org/packages/devel/bioc/vignettes/minfi/inst/doc/minfi.html
input data to this package are IDAT files

***Principe du methylation array***
450,000 CpG positions tested (out of 28 millions CpG)/ Each CpG is associated with two measurements: a methylated measurement and an “un”-methylated measurement

*These two values can be measured in one of two ways: using a “Type I” design or a “Type II design”. CpGs measured using a Type I design are measured using a single color, with two different probes in the same color channel providing the methylated and the unmethylated measurements. CpGs measured using a Type II design are measured using a single probe, and two different colors provide the methylated and the unmethylated measurements.
Practically, this implies that on this array there is not a one-to-one correspondence between probes and CpG positions. We have therefore tried to be precise about this and we refer to a “locus” (or “CpG”) when we refer to a single-base genomic locus, and we differentiate this from a “probe”. The previous generation 27k methylation array uses only the Type I design, and the EPIC arrays uses both Type I and Type II.
Differences in DNA methylation between samples can either be at a single CpG which is called a differentially methylated position (DMP), or at a regional level which is called a differentially methylated region (DMR)*


 
**Installation**
To install core packages, type the following in an R command window:

In [2]:
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version = "3.12")

Warning message:
“package ‘BiocManager’ was built under R version 4.0.5”
Bioconductor version '3.12' is out-of-date; the current release version '3.18'
  is available with R version '4.3'; see https://bioconductor.org/install

'getOption("repos")' replaces Bioconductor standard repositories, see
'?repositories' for details

replacement repositories:
    CRAN: https://cran.r-project.org


Bioconductor version 3.12 (BiocManager 1.30.16), R 4.0.3 (2020-10-10)

Installation paths not writeable, unable to update packages
  path: /shared/ifbstor1/software/miniconda/envs/r-4.0.3/lib/R/library
  packages:
    abcrf, acepack, ade4, adegenet, admisc, amap, apcluster, ape, argparse,
    aricode, ashr, askpass, babelgene, backports, base64, BAT, batchtools,
    bayesm, bayestestR, bbmle, bdsmatrix, beanplot, BH, BiasedUrn, BiocManager,
    bit, blob, bookdown, boot, brew, brio, broom, broom.helpers, bslib, cachem,
    Cairo, callr, car, carData, caret, changepoint, checkmate, circlize, class,
    

To install minfi package, start R (version "4.0.3") and enter:

In [3]:
BiocManager::install("minfi")

'getOption("repos")' replaces Bioconductor standard repositories, see
'?repositories' for details

replacement repositories:
    CRAN: https://cran.r-project.org


Bioconductor version 3.12 (BiocManager 1.30.16), R 4.0.3 (2020-10-10)

Warning message:
“package(s) not installed when version(s) same as current; use `force = TRUE` to
  re-install: 'minfi'”
Installation paths not writeable, unable to update packages
  path: /shared/ifbstor1/software/miniconda/envs/r-4.0.3/lib/R/library
  packages:
    abcrf, acepack, ade4, adegenet, admisc, amap, apcluster, ape, argparse,
    aricode, ashr, askpass, babelgene, backports, base64, BAT, batchtools,
    bayesm, bayestestR, bbmle, bdsmatrix, beanplot, BH, BiasedUrn, BiocManager,
    bit, blob, bookdown, boot, brew, brio, broom, broom.helpers, bslib, cachem,
    Cairo, callr, car, carData, caret, changepoint, checkmate, circlize, class,
    classInt, cli, clipr, clue, cluster, codetools, colorspace, colourpicker,
    commonmark, ComplexUpset, co

R sur JupyterLab vs.Rstudio?
Je veux utiliser le package Minfi pour analyser des données de Methylation array.
Il semblerait que j'aie besoin d'installer préalablement Bioconductor.
Pour R4.0.3(via JupyterLab)= version 3.12
Pour R4.1.1 (via Rstudio)= version 3.18
Quand j'essaie d'installer le package Minfi via BiocManager en utilisant un notebook Jupyter, j'obtiens un message d'erreur car version trop ancienne de plusieurs packages nécessaires =  ne peuvent plus être updatés.
Par contre, je peux loader les versions plus récentes de ces packages via BiocManager sur RStudio.
**Aide trouvée sur uilitR** 
*4.2.5 Mettre à jour les packages*
Il peut arriver qu’un de vos programmes génère une erreur parce que vous utilisez un package dans une version trop ancienne. Pour mettre à jour un package, il suffit de le réinstaller avec **install.packages**. Cette méthode de mise à jour est la plus simple, et résout la très grande majorité des problèmes de version de packages.
Il existe également la fonction **update.packages**. Cette fonction ne sert pas à mettre un package particulier, mais à mettre à jour tous les packages d’une librairie. Vous pouvez utiliser cette fonction si vous le souhaitez, mais son usage est plus complexe que celui d’install.packages. Vous pouvez en apprendre davantage sur cette fonction en consultant l’aide associée (?update.packages).


In [4]:
BiocManager::install("minfi")

'getOption("repos")' replaces Bioconductor standard repositories, see
'?repositories' for details

replacement repositories:
    CRAN: https://cran.r-project.org


Bioconductor version 3.12 (BiocManager 1.30.16), R 4.0.3 (2020-10-10)

Warning message:
“package(s) not installed when version(s) same as current; use `force = TRUE` to
  re-install: 'minfi'”
Installation paths not writeable, unable to update packages
  path: /shared/ifbstor1/software/miniconda/envs/r-4.0.3/lib/R/library
  packages:
    abcrf, acepack, ade4, adegenet, admisc, amap, apcluster, ape, argparse,
    aricode, ashr, askpass, babelgene, backports, base64, BAT, batchtools,
    bayesm, bayestestR, bbmle, bdsmatrix, beanplot, BH, BiasedUrn, BiocManager,
    bit, blob, bookdown, boot, brew, brio, broom, broom.helpers, bslib, cachem,
    Cairo, callr, car, carData, caret, changepoint, checkmate, circlize, class,
    classInt, cli, clipr, clue, cluster, codetools, colorspace, colourpicker,
    commonmark, ComplexUpset, co

In [5]:
update.packages

function (lib.loc = NULL, repos = getOption("repos"), contriburl = contrib.url(repos, 
    type), method, instlib = NULL, ask = TRUE, available = NULL, 
    oldPkgs = NULL, ..., checkBuilt = FALSE, type = getOption("pkgType")) 
{
    force(ask)
    text.select <- function(old) {
        update <- NULL
        for (k in seq_len(nrow(old))) {
            cat(old[k, "Package"], ":\n", "Version", old[k, "Installed"], 
                "installed in", old[k, "LibPath"], if (checkBuilt) 
                  paste("built under R", old[k, "Built"]), "\n", 
                "Version", old[k, "ReposVer"], "available at", 
                simplifyRepos(old[k, "Repository"], type))
            cat("\n")
            answer <- askYesNo("Update?")
            if (is.na(answer)) {
                cat("cancelled by user\n")
                return(invisible())
            }
            if (isTRUE(answer)) 
                update <- rbind(update, old[k, ])
        }
        update
    }
    if (is.null(lib.loc)) 
        lib.loc <- .libPaths()
    if (type == "both" && (!missing(contriburl) || !is.null(available))) {
        stop("specifying 'contriburl' or 'available' requires a single type, not type = \"both\"")
    }
    if (is.null(available)) {
        available <- available.packages(contriburl = contriburl, 
            method = method, ...)
        if (missing(repos)) 
            repos <- getOption("repos")
    }
    if (!is.matrix(oldPkgs) && is.character(oldPkgs)) {
        subset <- oldPkgs
        oldPkgs <- NULL
    }
    else subset <- NULL
    if (is.null(oldPkgs)) {
        oldPkgs <- old.packages(lib.loc = lib.loc, contriburl = contriburl, 
            method = method, available = available, checkBuilt = checkBuilt)
        if (missing(repos)) 
            repos <- getOption("repos")
        if (!is.null(oldPkgs)) {
            pkg <- 0L
            while (pkg < nrow(oldPkgs)) {
                pkg <- pkg + 1L
                if (find.package(oldPkgs[pkg], lib.loc = lib.loc) != 
                  find.package(oldPkgs[pkg], lib.loc = oldPkgs[pkg, 
                    2])) {
                  warning(sprintf("package '%s' in library '%s' will not be updated", 
                    oldPkgs[pkg], oldPkgs[pkg, 2]), call. = FALSE, 
                    immediate. = TRUE)
                  oldPkgs <- oldPkgs[-pkg, , drop = FALSE]
                  pkg <- pkg - 1L
                }
            }
        }
        if (is.null(oldPkgs)) 
            return(invisible())
    }
    else if (!(is.matrix(oldPkgs) && is.character(oldPkgs))) 
        stop("invalid 'oldPkgs'; must be a character vector or a result from old.packages()")
    if (!is.null(subset)) {
        oldPkgs <- oldPkgs[rownames(oldPkgs) %in% subset, , drop = FALSE]
        if (nrow(oldPkgs) == 0) 
            return(invisible())
    }
    update <- if (is.character(ask) && ask == "graphics") {
        if (.Platform$OS.type == "windows" || .Platform$GUI == 
            "AQUA" || (capabilities("tcltk") && capabilities("X11"))) {
            k <- select.list(oldPkgs[, 1L], oldPkgs[, 1L], multiple = TRUE, 
                title = "Packages to be updated", graphics = TRUE)
            oldPkgs[match(k, oldPkgs[, 1L]), , drop = FALSE]
        }
        else text.select(oldPkgs)
    }
    else if (isTRUE(ask)) 
        text.select(oldPkgs)
    else oldPkgs
    if (length(update)) {
        if (is.null(instlib)) 
            instlib <- update[, "LibPath"]
        libs <- unique(instlib)
        for (l in libs) if (type == "both") 
            install.packages(update[instlib == l, "Package"], 
                l, repos = repos, method = method, ..., type = type)
        else install.packages(update[instlib == l, "Package"], 
            l, contriburl = contriburl, method = method, available = available, 
            ..., type = type)
    }
}
<bytecode: 0x55a374502c98>
<environment: namespace:utils>

Chargement de minfi

In [6]:
library(minfi)

Loading required package: BiocGenerics

Loading required package: parallel


Attaching package: ‘BiocGenerics’


The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB


The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs


The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min


Loading required package: GenomicRanges

Loading required package: stats4

Loading required package: S4Vectors


At

**Récupération des data à partir de GEOquery**
charger GEOquery pour importer les datas publiques d'une étude 

In [7]:
library(GEOquery)

si données dans supplemental files

In [8]:
getGEOSuppFiles("GSE137716")

size     
/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 472432640
                                                                                                   isdir
/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar FALSE
                                                                                                   mode
/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 660 
                                                                                                   mtime              
/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 2023-11-13 12:18:28
                                                                                                   ctime              
/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 2023-11-13 12:18:28
                                                                                                   atime              
/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 2023-11-10 16:33:42
                                                                                                   uid   
/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 163697
                                                                                                   gid   
/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar 163697
                                                                                                   uname  
/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar sletuve
                                                                                                   grname 
/shared/ifbstor1/projects/202304_duo/sletuve/Projet/Projet-Methylation/GSE137716/GSE137716_RAW.tar sletuve

décompresser archive qui contient les idat files (_RAW) et extraire dans dossier exdir

In [9]:
untar("GSE137716/GSE137716_RAW.tar", exdir = "GSE137716/idat")

On voit ici qu'il y a des files Grn (green) et Red, 2 files pour chaque échantillon

In [10]:
head(list.files("GSE137716/idat", pattern = "idat"))

[1] "GSM4085551_9976861129_R03C02_Grn.idat.gz"
[2] "GSM4085551_9976861129_R03C02_Red.idat.gz"
[3] "GSM4085552_9979858039_R01C01_Grn.idat.gz"
[4] "GSM4085552_9979858039_R01C01_Red.idat.gz"
[5] "GSM4085553_9976861137_R03C02_Grn.idat.gz"
[6] "GSM4085553_9976861137_R03C02_Red.idat.gz"

décompresser les idat files

In [11]:
idatFiles <- list.files("GSE68777/idat", pattern = "idat.gz$", full = TRUE)

In [12]:
sapply(idatFiles, gunzip, overwrite = TRUE)

named list()

Lire les files

In [13]:
rgSet <- read.metharray.exp("GSE137716/idat")

ERROR: Error in read.metharray.exp("GSE137716/idat"): No IDAT files were found
